app/controllers/IncomeController.scala

```diff
@@ -140,35 +142,34 @@ class IncomeController @Inject()(
     })
   }
 
-  def editRegularIncome(): Action[AnyContent] = (authenticate andThen validatePerson).async { implicit request =>
-    implicit val user: AuthedUser = request.taiUser
-
-    journeyCacheService
-      .collectedJourneyValues(
-        Seq(UpdateIncome_PayToDateKey, UpdateIncome_IdKey, UpdateIncome_NameKey),
-        Seq(UpdateIncome_DateKey))
-      .getOrFail flatMap {
-      case (mandatorySeq, optionalSeq) =>
-        val date = Try(optionalSeq.head.map(date => LocalDate.parse(date))) match {
-          case Success(optDate) => optDate
-          case Failure(exception) =>
-            logger.warn(s"Unable to parse updateIncomeDateKey  $exception")
-            None
-        }
-
-        val employerName = mandatorySeq(2)
-        val payToDate = BigDecimal(mandatorySeq.head)
+  def editRegularIncome(empId: Int): Action[AnyContent] = (authenticate andThen validatePerson).async {
```

Comment:
I find this very hard to follow. What happens form a high level when the user submits and they then click back? Have UX been asked to see what is best? As we are now using mandatoryJourneyValueAsInt to get from the cache, we can handle a failing retrieval from the cache better using the either, and maybe send them back to the start of the journey which is what is done elsewhere in the service? Do we need to be letting the user submit again?
Here is an example of how I've seen it done before 
https://github.com/hmrc/tai-frontend/blob/afff5a0634235db1fd151f3bba0175afff789c6d/app/controllers/income/UpdateIncomeNextYearController.scala#L256